### PR TITLE
[Bug]: system table, panic info occured #5892

### PR DESCRIPTION
**Pull request by badboynt1**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #5892

## What this PR does / why we need it:
initialize null bitmap correctly when reading from external table.

### CODE REVIEW

**Review by fengttt — CHANGES_REQUESTED** [review]
@nnsgmsone @m-schen please take a look at line 256 - 260.

What is happening?   We create a new orig vector, prealloc, then set original to false, and then direclty manipulate Nsp?

File: pkg/sql/colexec/external/external.go [hunk 1]
```diff
@@ -256,6 +257,7 @@ func makeBatch(param *ExternalParam, plh *ParseLineHandler, mp *mpool.MPool) *ba
 		vec := vector.NewOriginal(typ)
 		vector.PreAlloc(vec, batchSize, batchSize, mp)
 		vec.SetOriginal(false)
```
Comment (nnsgmsone) [inline pkg/sql/colexec/external/external.go:259]: What is this going to do....

**Review by nnsgmsone — CHANGES_REQUESTED** [review]

File: pkg/sql/colexec/external/external.go (hunk 1 above)
Comment (m-schen) [reply pkg/sql/colexec/external/external.go:259]: just use vector.NewWithNspSize() instead of line 257 and 259 and 260.

Comment (badboynt1) [reply pkg/sql/colexec/external/external.go:258]: > 

fixed

File: pkg/sql/colexec/external/external.go [hunk 2]
```diff
@@ -253,7 +253,7 @@ func makeBatch(param *ExternalParam, plh *ParseLineHandler, mp *mpool.MPool) *ba
 	//alloc space for vector
 	for i := 0; i < len(param.Attrs); i++ {
 		typ := types.New(types.T(param.Cols[i].Typ.Id), param.Cols[i].Typ.Width, param.Cols[i].Typ.Scale, param.Cols[i].Typ.Precision)
-		vec := vector.NewOriginal(typ)
+		vec := vector.NewWithNspSize(typ, int64(param.batchSize))
 		vector.PreAlloc(vec, batchSize, batchSize, mp)
 		vec.SetOriginal(false)
```
Comment (fengttt) [inline pkg/sql/colexec/external/external.go:258]: I don't think you need SetOriginal

Comment (badboynt1) [reply pkg/sql/colexec/external/external.go]: > I don't think you need SetOriginal

OK， fixed

File: pkg/sql/colexec/external/external.go [hunk 3]
```diff
@@ -253,9 +253,8 @@ func makeBatch(param *ExternalParam, plh *ParseLineHandler, mp *mpool.MPool) *ba
 	//alloc space for vector
 	for i := 0; i < len(param.Attrs); i++ {
 		typ := types.New(types.T(param.Cols[i].Typ.Id), param.Cols[i].Typ.Width, param.Cols[i].Typ.Scale, param.Cols[i].Typ.Precision)
-		vec := vector.NewOriginal(typ)
+		vec := vector.NewWithNspSize(typ, int64(param.batchSize))
 		vector.PreAlloc(vec, batchSize, batchSize, mp)
```
Comment (nnsgmsone) [inline pkg/sql/colexec/external/external.go:257]: Why do you want to use it that way.

**Review by nnsgmsone — CHANGES_REQUESTED** [review]

**Review by nnsgmsone — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

File: pkg/container/nulls/nulls.go [hunk 4]
```diff
@@ -239,7 +248,11 @@ func Filter(n *Nulls, sels []int64) *Nulls {
 		sp = unsafe.Slice((*uint64)(unsafe.Pointer(&sels[0])), cap(sels))[:len(sels)]
 	}
 	np := bitmap.New(len(sels))
+	upperLimit := uint64(n.Np.Len())
 	for i, sel := range sp {
+		if sel >= upperLimit {
```
Comment (fengttt) [inline pkg/container/nulls/nulls.go:253]: Do you know why this ever happens?

**Review by fengttt — APPROVED** [review]

**Pull request merged**